staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -879,6 +894,10 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 		handler = genericfilters.WithMaxInFlightLimit(handler, c.MaxRequestsInFlight, c.MaxMutatingRequestsInFlight, c.LongRunningFunc)
 	}
 
+	if len(c.NotReadyDebuggerGroup) > 0 {
+		handler = genericfilters.WithNotReady(handler, c.lifecycleSignals.HasBeenReady.Signaled(), c.NotReadyDebuggerGroup)
+	}
```

Comment:
I like the idea of the new filter being gated, but since it shouldn't be used in normal OCP use cases, I wouldn't want OCP users to be able to enable it.

Instead, what do you think about adding a new [ControlPlaneTopology](https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L84-L92) for Microshift and gate the Microshift-specific code behind it? This is the current approach that we have in OCP for SNO and I think it would kind of make sense to extend it for Microshift.

Reply:
that's ideal, also depends on how microshift sets up kube-apiserver, from current setup, looks like we need a configurable apiserver argument.  https://github.com/openshift/microshift/pull/924 

if there is a way to hook into the `server.Config` struct, then we could remove the server option and just do with the bool member of the `Config` struct.